File: src/main/java/chat/teco/tecochat/chat/domain/chat/QuestionAndAnswer.java
package chat.teco.tecochat.chat.domain.chat;

import static java.util.Objects.requireNonNull;

import chat.teco.tecochat.common.entity.BaseEntity;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;

@Entity
public class QuestionAndAnswer extends BaseEntity {

    @Embedded
    private Question question;

    @Embedded
    private Answer answer;

    protected QuestionAndAnswer() {
    }

    public QuestionAndAnswer(String question, String answer) {
        this(Question.question(question), Answer.answer(answer));
    }

    public QuestionAndAnswer(Question question, Answer answer) {
        this.question = requireNonNull(question);
        this.answer = requireNonNull(answer);
    }

    public QuestionAndAnswer copy() {
        return new QuestionAndAnswer(question, answer);
    }

    public Question question() {
        return question;
    }

    public Answer answer() {
        return answer;
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/chat/QuestionAndAnswer.kt
package chat.teco.tecochat.domain.chat

import chat.teco.tecochat.support.domain.BaseEntity
import jakarta.persistence.Embedded
import jakarta.persistence.Entity

@Entity
class QuestionAndAnswer(

    @Embedded
    val question: Question,

    @Embedded
    val answer: Answer,

    id: Long = 0L,
) : BaseEntity(id) {

    constructor(question: String, answer: String) : this(
        Question.question(question), Answer.answer(answer)
    )

    fun copy(): QuestionAndAnswer {
        return QuestionAndAnswer(question, answer)
    }
}
